### PR TITLE
Add political checkbox to editions

### PR DESCRIPTION
**Pull request by edds**
So that we can manually change the political state of an edition add a
checkbox to the admin. This will be useful if, for example, a document
which doesn't meet the standard criteria for being political is noticed
to be political. Only managing editors and GDS staff should be able to
change the state of the political flag. The political checkbox should
only be shown after the document has been published for the first time.
This is so that we can automatically set it on first publish. If it was
editable before the first publish and editor changed it our automatic
political marking would overwrite what the editor had set.

screen shot 2015-03-03 at 14 05 05

---

This still needs the words next to the checkbox checking and either this
or https://github.com/alphagov/whitehall/pull/1991 need to remove the database migration depending on which gets
merged first

### CODE REVIEW

**Comment by tekin** [comment]
Would be good to see some tests around the authority rules here. Some unit tests verifying that only managing editors can change the political designation of editions will suffice.

**Comment by fofr** [comment]
It'd be good to see some screenshots as to how this looks.

**Comment by edds** [comment]
Added a unit test for each level of user type. Added a screenshot to the PR description.

File: app/views/admin/editions/_standard_fields.html.erb [hunk 1]
```diff
@@ -14,6 +14,12 @@
   <%= form.text_area :body, class: "previewable", rows: 20, cols: 9000, required: form.object.body_required? %>
 
   <%= render partial: "additional_significant_fields", locals: {form: form, edition: form.object} %>
+
+  <% if form.object.previously_published and can?(:mark_political, edition) %>
```
Comment (tekin) [inline app/views/admin/editions/_standard_fields.html.erb]: I don't believe `previously_published` will behave as you want here. If a user sets the `first_published_at` on a brand new document, then `previously_published` will return true and the checkbox will be displayed on subsequent edits, even though the document hasn't actually been published on GOV.UK yet.

Comment (tekin) [inline app/views/admin/editions/_standard_fields.html.erb]: Try `if edition.document.published?` instead.

File: test/functional/admin/generic_editions_controller_tests/political_document_test.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class Admin::GenericEditionsController::PolticalDocumentsTest < ActionController::TestCase
+  tests Admin::NewsArticlesController
+
+  setup do
+    login_as :policy_writer
+  end
+
+  view_test "displays the political checkbox for privileged users " do
+    login_as :managing_editor
+    edition = create(:news_article, first_published_at: 2.days.ago)
```
Comment (tekin) [inline test/functional/admin/generic_editions_controller_tests/political_document_test.rb]: These tests will want updating too: You'll need to create a published edition and then a subsequent draft from that to simulate an actual edition that has been previously published. Something like this:
```
published_edition = create(:published_news_article)
new_draft = create(:news_article, document: published_edition.document)
```

Comment (edds) [inline test/functional/admin/generic_editions_controller_tests/political_document_test.rb]: Good catch. Updated.

**Comment by tekin** [comment]
Assuming a green build: :+1:

**Pull request merged**